### PR TITLE
Action/Generate: Best effort warnings when flags are ignored

### PR DESCRIPTION
**Pull request by Profpatsch**
The command line parser accepts any flags in `generate`, but no warning is shown if user instructions are ignored.

This is a best-effort way of warning if a flag would be ignored, making it more user-friendly.

Ideally, this would be done before starting work, but I don’t see a good way to patch it into the command line parser; maybe by having an intermediate parsing step that parses into an ADT first and throws warnings for ignored arguments.

Another case I saw is that “--haskell” and “--frege” can both be set, but “--haskell” will always be ignored.

### CODE REVIEW

File: src/Action/Generate.hs [hunk 1]
```diff
@@ -201,17 +201,26 @@ actionGenerate g@Generate{..} = withTiming (if debug then Just $ replaceExtensio
     createDirectoryIfMissing True $ takeDirectory database
     whenLoud $ putStrLn $ "Generating files to " ++ takeDirectory database
 
+    let warnFlagIgnored thisFlag reason ignoredFlagPred ignoredFlag =
+          when ignoredFlagPred $ putStrLn $ "Warning: " <> thisFlag <> " is " <> reason <> ", which means " <> ignoredFlag <> " is ignored."
```
Comment (ndmitchell) [inline src/Action/Generate.hs:205]: Everywhere else in this file uses ++ for string concat

**Comment by ndmitchell** [comment]
I think parsing into an ADT would be better, but this seems a step in the right direction.

File: src/Action/Generate.hs (hunk 1 above)
Comment (Profpatsch) [reply src/Action/Generate.hs:205]: Ah, sorry, the power of habit

Comment (ndmitchell) [reply src/Action/Generate.hs:205]: No big deal - not worth fussing about either way

**Pull request merged**